libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/MapboxCarNavigationManager.kt

```diff
@@ -42,19 +43,26 @@ class MapboxCarNavigationManager internal constructor(
         onUpdateTrip(trip)
     }
 
-    private val tripSessionStateObserver = TripSessionStateObserver { tripSessionState ->
-        logAndroidAuto("$LOG_CATEGORY tripSessionStateObserver state: $tripSessionState")
-        when (tripSessionState) {
-            TripSessionState.STARTED -> navigationManager.navigationStarted()
-            TripSessionState.STOPPED -> navigationManager.navigationEnded()
+    private var inActiveNavigation = false
+
+    private val routesObserver = RoutesObserver {
+        if (it.navigationRoutes.isEmpty()) {
```

Comment:
Should we also check `inActiveNavigation` here to avoid calling `navigationManager.navigationEnded()` if it is already ended?